### PR TITLE
Upgrade  for Arduino-Espressif32 2.0.x

### PR DESCRIPTION
**Pull request by FernandoGarcia**
While 2.0.x is not available in PIO let's use Tasmota repository.

I have tested this code with NodeMCU and ESP32-Wroover IE.

In This PR I haven't changed the file structure.

### CODE REVIEW

**Comment by raomin** [comment]
FWIT I support this PR. I tested it on a clean clone + default framework (2.0.2) + esp32, it compiles and works.
Maybe we could condition the changes not to break v1.0.6 with
```c
#ifdef ESP_IDF_VERSION_MAJOR 
``` 

I don't know how other projects/libs are dealing with this but supporting v2.0 is really needed now.

**Comment by rjwats** [comment]
Yes agreed, has anyone tested Espressif32 @ 4.3.0 with these changes?

**Comment by rjwats** [comment]
https://github.com/me-no-dev/ESPAsyncWebServer/issues/1151

Quickly tested, it looks like this issue is affecting the project when upgrading the framework version to 4.3.0.

**Comment by proddy** [comment]
we've been testing 4.3.0 on my project and ran into some snags around Wifi and AP (https://github.com/emsesp/EMS-ESP32/issues/473)

**Comment by rjwats** [comment]
Incorperated changes from here into platform/react upgrade branch:

https://github.com/rjwats/esp8266-react/pull/301/files

File: lib/framework/WiFiSettingsService.cpp [hunk 1]
```diff
@@ -71,10 +71,10 @@ void WiFiSettingsService::manageSTA() {
     } else {
       // configure for DHCP
 #ifdef ESP32
-      WiFi.config(INADDR_NONE, INADDR_NONE, INADDR_NONE);
+      //WiFi.config(INADDR_NONE, INADDR_NONE, INADDR_NONE);
       WiFi.setHostname(_state.hostname.c_str());
 #elif defined(ESP8266)
-      WiFi.config(INADDR_ANY, INADDR_ANY, INADDR_ANY);
+      //WiFi.config(INADDR_ANY, INADDR_ANY, INADDR_ANY);
```
[the remark this answers is not in the document]
Comment (rjwats) [reply lib/framework/WiFiSettingsService.cpp:77]: I'm not seeing this issue with espressif's platform (version 4.3.0) - see other open pr and this branch:

https://github.com/rjwats/esp8266-react/tree/upgrade-e32-4.3.0

**Comment by rjwats** [comment]
4.3.0 support added

File: lib/framework/WiFiSettingsService.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (FernandoGarcia) [reply lib/framework/WiFiSettingsService.cpp:77]: Hi!
Using your latest version I have the problem with wrong IP as pointed [here](https://github.com/rjwats/esp8266-react/issues/285#issuecomment-1077839979).

Captura de tela de 2022-05-21 14 15 40

The problem is caused by this line as I told before.
My board is ESP32-Wrover IE.
Best regards.

**Comment by rjwats** [comment]
hummm... I can't seem to re-create that, what connection setup are you
using?

**Comment by FernandoGarcia** [comment]
I'm using default settings for network connection.

I have added just this env to .ini file.

```
[env:esp32dev]
; Comment out min_spiffs.csv setting if disabling PROGMEM_WWW with ESP32
board_build.partitions = min_spiffs.csv
platform = espressif32
board = esp32dev
board_build.filesystem = littlefs
```

**Pull request closed without merging**